data/transactions/logic/eval.go

```diff
@@ -324,6 +339,7 @@ type EvalParams struct {
 
 	// readBudgetChecked allows us to only check the read budget once
 	readBudgetChecked bool
+	SurplusReadBudget uint64
```

Comment:
minor: could you add a comment for `SurplusReadBudget`, though I believe this is just representing the remainder of read budget throughout evaluation?

Reply:
👍  see 79b3818b0